sdk/core/core-lro/src/http/operation.ts

```diff
@@ -304,15 +304,20 @@ export function getOperationStatus<TState>(
   }
 }
 
+function accessBody<P extends string>(
+  { flatResponse, rawResponse }: LroResponse,
+  prop: P
+): ResponseBody[P] {
+  return (flatResponse as ResponseBody)?.[prop] ?? (rawResponse.body as ResponseBody)?.[prop];
+}
+
 export function getResourceLocation<TState>(
-  { flatResponse }: LroResponse,
+  res: LroResponse,
   state: RestorableOperationState<TState>
 ): string | undefined {
-  if (typeof flatResponse === "object") {
-    const resourceLocation = (flatResponse as { resourceLocation?: string }).resourceLocation;
-    if (resourceLocation !== undefined) {
-      state.config.resourceLocation = resourceLocation;
-    }
+  const loc = accessBody(res, "resourceLocation");
+  if (loc !== undefined) {
+    state.config.resourceLocation = loc;
```

Comment:
Just confirm - does that mean the body resourceLocation would override the existing `state.config.resourceLocation` ?

Reply:
Standard LRO flow doesn't have a case where resource location can be specified in different places. It is typically either in the initial polling response in the location header or in a resourceLocation property in the polling response. Is there a specific scenario in mind where resource location could be specified in two different places and not identical?

Reply:
Not yet I think, will monitor this!